### PR TITLE
Add C++ Solution + Editorial to Mortal Combat Tower (Gold)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
imma throw the next contest...

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by ryanchou-dev** [comment]
also, any thoughts on this section?

it says introductory usaco, but it has two non-usaco tasks
image

**Comment by maggie-j-liu** [comment]
maybe rename the section to Introductory Problems?

File: solutions/cf-1418C.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+void solve() {
```
Comment (SansPapyrus683) [inline solutions/cf-1418C.mdx:38]: put this in main

File: solutions/cf-1418C.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		}
+	}
+	cout << min(dp[0][n], dp[1][n]) << '\n';
+}
+
+int main() {
+	int t;
+	cin >> t;
+
+	while (t--) {
```
Comment (SansPapyrus683) [inline solutions/cf-1418C.mdx:77]: (optional) use for loop

File: solutions/cf-1418C.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+void solve() {
+	int n;
+	cin >> n;
+
+	vector<int> a(n);
```
Comment (SansPapyrus683) [inline solutions/cf-1418C.mdx:42]: (optional) rename to `bosses`

File: solutions/cf-1418C.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	for (int i = 0; i < n; i++) {
+		cin >> a[i];
+	}
+
+	/*
+	 * dp[i][j] = the min amt of skip points 
+	 * on turn i (your turn is 0), on the jth boss.
+	 */
+	vector<vector<int>> dp(2, vector<int>(n + 1, 1e9));
```
Comment (SansPapyrus683) [inline solutions/cf-1418C.mdx:51]: use `INT32_MAX`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-1418C.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/cf-1418C.mdx:51]: wouldn't that overflow when doing the addition?

Comment (SansPapyrus683) [reply solutions/cf-1418C.mdx]: hm
add a comment, `1e9 to prevent overflow`

File: solutions/cf-1418C.mdx [hunk 5]
```diff
@@ -35,47 +35,43 @@ So, the answer would be $\min(dp[N][0], dp[N][1])$.
...
+			dp[0][i + 1] = min(dp[0][i + 1], dp[1][i] + bosses[i]);
+			dp[1][i + 1] = min(dp[1][i + 1], dp[0][i]);
+
+			// the opposite player switches from the previous two moves.
+			if (i + 2 <= n) {
+				dp[0][i + 2] = min(dp[0][i + 2], dp[1][i] + bosses[i] + bosses[i + 1]);
+				dp[1][i + 2] = min(dp[1][i + 2], dp[0][i]);
+			}
+		}
+		cout << min(dp[0][n], dp[1][n]) << '\n';
```
Comment (SansPapyrus683) [inline solutions/cf-1418C.mdx:74]: (optional) use `endl` if it runs fine w/o

File: solutions/cf-1418C.mdx [hunk 6]
```diff
@@ -71,7 +71,7 @@ int main() {
 				dp[1][i + 2] = min(dp[1][i + 2], dp[0][i]);
 			}
 		}
-		cout << min(dp[0][n], dp[1][n]) << '\n';
+		cout << min(dp[0][n], dp[1][n]) << endl;
```
Comment (SansPapyrus683) [inline solutions/cf-1418C.mdx:74]: wow can't believe you actually did it

File: solutions/cf-1418C.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+$\texttt{dp}[i + 2][0] =\min(\texttt{dp}[i + 2][0], \texttt{dp}[i][1] + \texttt{hard}[i] + \texttt{hard}[i + 1])$
+
+$\texttt{dp}[i + 1][1] =\min(\texttt{dp}[i + 1][1], \texttt{dp}[i][0])$
+
+$\texttt{dp}[i + 2][1] =\min(\texttt{dp}[i + 2][1], \texttt{dp}[i][0])$
+
+This is because from each state, either the previous move, or the previous two moves could have been from the other player.
+
+So, the answer would be $\min(dp[N][0], dp[N][1])$.
```
Comment (dongliu0426) [inline solutions/cf-1418C.mdx:25]: textttify "dp"

File: solutions/cf-1418C.mdx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+		int n;
+		cin >> n;
+
+		vector<int> bosses(n);
+		for (int i = 0; i < n; i++) {
+			cin >> bosses[i];
+		}
+
+		/*
+		 * dp[i][j] = the min amt of skip points
```
Comment (dongliu0426) [inline solutions/cf-1418C.mdx:52]:
```suggestion
		 * dp[i][j] = the min amount of skip points 
```

Comment (SansPapyrus683) [reply solutions/cf-1418C.mdx:52]: (optional imo)

**Review by SansPapyrus683 — APPROVED** [review]

**Review by nwatx — APPROVED** [review]
code and explanation

**Review by TruaShamu — APPROVED** [review]
looks good to me :)

**Pull request merged**